src/cmdstan/command.hpp

```diff
@@ -644,50 +649,40 @@ int command(int argc, const char *argv[]) {
       throw std::invalid_argument(msg.str());
     }
 
-    size_t c_params_vec_size = 0;
-    size_t c_params_size = 0;
-    std::vector<double> c_params_r;
-    std::vector<size_t> dims_c_params_r;
+    int cpars_size = !(cpars_file->is_default()) ? 1 : 0;
+    // Store in single nested array to allow single loop for calc and print
+    size_t num_par_sets = u_params_vec_size + cpars_size;
+    std::vector<std::vector<double>> params_r_ind(num_par_sets);
+    std::vector<int> dummy_params_i;
     if (!(cpars_file->is_default())) {
       std::string c_fname(cpars_file->value());
       std::ifstream c_stream(c_fname.c_str());
 
       std::shared_ptr<stan::io::var_context> cpars_context
           = get_var_context(c_fname);
-
-      c_params_r = (*cpars_context).vals_r("params_r");
-      dims_c_params_r = (*cpars_context).dims_r("params_r");
-
-      c_params_vec_size = dims_c_params_r.size() == 2 ? dims_c_params_r[0] : 1;
-      c_params_size = dims_c_params_r.size() == 2 ? dims_c_params_r[1]
-                                                  : dims_c_params_r[0];
-    }
-
-    if (c_params_size > 0 && c_params_size != param_names.size()) {
-      msg << "Incorrect number of constrained parameters provided! "
-             "Model has "
-          << param_names.size() << " parameters but " << c_params_size
-          << " were found.";
-      throw std::invalid_argument(msg.str());
+      std::vector<std::string> input_cpar_names;
+      (*cpars_context).names_r(input_cpar_names);
+      for (std::string &m_param_name : param_names) {
+        bool present
+            = std::any_of(input_cpar_names.begin(), input_cpar_names.end(),
+                          [m_param_name](std::string &c_param_name) {
+                            return m_param_name == c_param_name;
+                          });
+
+        if (!present) {
+          msg << "Constrained value(s) for parameter " << m_param_name
+              << " not found!";
+          throw std::invalid_argument(msg.str());
+        }
+      }
+      model.transform_inits((*cpars_context), dummy_params_i, params_r_ind[0],
```

Comment:
How do we know `params_r_ind` is non-empty so that indexing won't cause a segfault?

Reply:
The `params_r_ind` vector is initialised to the number of parameter sets passed in (both constrained and unconstrained), if no parameter sets are passed then the program throws before it gets to this indexing